lib/azure/armrest/storage_account_service.rb

```diff
@@ -283,20 +283,26 @@ def get_private_images(storage_accounts)
           storage_account.containers.each do |container|
             next if container.name =~ /^bootdiagnostics/i
             storage_account.blobs(container.name).each do |blob|
-              next unless File.extname(blob.name).casecmp('.vhd').zero?
-              next unless blob.properties.lease_state.casecmp('available').zero?
+              next unless File.extname(blob.name_from_hash).casecmp('.vhd').zero?
+              next unless blob.lease_state_from_hash.casecmp('available').zero?
 
               # In rare cases the endpoint will be unreachable. Warn and move on.
               begin
                 blob_properties = storage_account.blob_properties(blob.container, blob.name)
+                blob_properties = storage_account.blob_properties(
```

Comment:
Oh man... I suck at rebasing... the line above this needs to be removed (since it is a dup).  Thankfully my benchmarking script caught this.

Will remove that shortly.